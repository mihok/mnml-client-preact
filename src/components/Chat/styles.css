/* Chat Component */


/* Messenger Theme */

.Chat--messenger {
  display: flex;
  flex-direction: column;
  background: rgba(248,248,248,1);
  flex-grow: 1;
  box-shadow: #dddddd 1px 1px 8px 0;
  border: 0; 
  border-radius: 0 3px 3px 0;
}

.Chat__body--messenger {
  list-style: none;
  display: flex;
  flex-grow: 1;
  padding: 6px 4px;
  flex-direction: column;
  overflow-y: scroll;
  margin: 0;
  border-right: 1px solid rgba(0,0,0,0.1);
}


/* Float Theme */

.Chat--float {
  position: absolute;
  right: 30px;
  bottom: 60px;
  background-color: transparent;
}

.Chat__body--float {
  display: block;
  width: 280px;
  height: 80vh;
  overflow: scroll;
  display: flex;
  flex-direction: column;
  list-style: none;
  flex-grow: 1;
  justify-content: flex-end;
  padding-bottom: 64px;
  /* Push last message down; gives <Input /> room to breathe */
}


