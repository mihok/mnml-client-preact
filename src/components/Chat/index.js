/**
 * Main chat component handles displaying chat messages and passes
 * functions to the input to send messages
 * Has scroll to bottom functionality for keeping window scoll at bottom of the chat.
 * Needs to be a class based component in order to have that functionality ^
 */

import { h, render, Component } from 'preact';
import './styles.css';
import Message from '../Message/';
import themer from '../ThemeProvider/themeHOC';
import Input from '../Input/index';

class Chat extends Component {
  componentDidMount () {
    this.scrollToBottom();
  }

  componentDidUpdate () {
    this.scrollToBottom();
  }

  scrollToBottom = () => {
    this.container.scrollTop = this.container.scrollHeight;
  };

  renderMessages = () => this.props.messages.map(msg => <Message type={msg.author} content={msg.content} />, // msg.content is an array!
    );

  render () {
    const {
      toggleChat,
      textBox,
      handleInput,
      sendMessage,
      theme,
    } = this.props;

    return (
      <section className={`Chat__${theme}`}>

        <header className={`Chat__Header__${theme}`}>
          <div className={`Chat__OperatorName__${theme}`}>Chat with John</div>
          <button
            className={`Chat__CloseBtn__${theme}`}
            onClick={() => toggleChat(false)}
          >
            x{' '}
          </button>
        </header>

        {/* Container for text input and reading messages */}
        <ul className={`Chat__Body__${theme}`} ref={c => (this.container = c)}>
          {this.renderMessages()}
        </ul>

        <Input
          sendMessage={sendMessage}
          textBox={textBox}
          handleInput={handleInput}
        />

        {/*
        <form class={`Chat__Form`} onSubmit={sendMessage}>
          <input
            class={`Chat__Input__${theme}`}
            placeholder="Type Here"
            onChange={e => handleInput(e)}
            name="messages"
            value={textBox}
          />
        </form>
        */}

      </section>
    );
  }
}

export default themer(Chat);
