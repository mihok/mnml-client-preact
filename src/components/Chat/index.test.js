/* const render = require('preact-render-to-string');
 * const App = require('../thing');*/
import render from 'preact-render-to-string';
import { h } from 'preact';
import Chat from '.';

describe('App component', () => {
  it('should render ', () => {
    const props = {
      messages: [],
    };

    const tree = render(<Chat messages={props.messages} />);
    expect(tree).toMatchSnapshot();
  });
});
