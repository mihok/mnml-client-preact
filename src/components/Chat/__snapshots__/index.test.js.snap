// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render  1`] = `"<section class=\\"Chat__undefined\\"><header class=\\"Chat__Header__undefined\\"><div class=\\"Chat__OperatorName__undefined\\">Chat with John</div><button class=\\"Chat__CloseBtn__undefined\\">x </button></header><ul class=\\"Chat__Body__undefined\\"></ul><form class=\\"Input__Form__undefined\\"><input class=\\"Input__undefined\\" placeholder=\\"Type Here\\" name=\\"messages\\" /></form></section>"`;
